sympy/solvers/diophantine.py

```diff
@@ -875,9 +875,8 @@ def _diop_quadratic(var, coeff, t):
                     + (sqa*g*u**2 + D*u + sqa*F) // _c
 
                 for z0 in range(0, abs(_c)):
-                    if divisible(
-                            sqa*g*z0**2 + D*z0 + sqa*F,
-                            _c):
+                    if (divisible(sqa*g*z0**2 + D*z0 + sqa*F, _c) and
```

Comment:
Factoring the expression will definitely make the code run faster. Thanks for the tip! I'll fix it as soon as I get my hands on an Internet connected computer.

This condition was not mentioned in the explanatory documentation. I derived it myself after substituting y with an expression in x as follows:

Since `u = (sqa*x + e*sqc*y)` we replace `y` with `((u - sqa*x)/(e*sqc))` and set x to be an integer to get the second condition I wrote. If you have more concerns, I'll be glad to show you my notes : smiley: